helix-term/src/commands.rs

```diff
@@ -3905,23 +3910,29 @@ fn remove_primary_selection(cx: &mut Context) {
 }
 
 pub fn completion(cx: &mut Context) {
+    // TODO completion starts to get ugly...
+    // maybe think about something like completion provider and separate completion-state into helix-view?
+    let clear_completion = async move {
+        let call = job::Callback::EditorCompositor(Box::new(
+            move |editor: &mut Editor, compositor: &mut Compositor| {
+                let ui = compositor.find::<ui::EditorView>().unwrap();
+                ui.clear_completion(editor);
+            },
+        ));
+        Ok(call)
+    };
+    cx.jobs.callback(clear_completion);
+
+    if let Err(e) = helix_lsp::block_on(cx.jobs.finish(cx.editor, None)) {
+        return cx.editor.set_status(format!("{e}"));
+    };
+
```

Comment:
> When using completion menu Ctrl+n/p, sometimes helix is blocked(any input)

Well I'm not sure where the error originates, but I suspect it doesn't have to do with this PR (can you try the same config on master?).

Regarding the blocking, can you try to comment out this block? If this helps, I have to think about a different way to sequentialize the LSP requests

Reply:
> > When using completion menu Ctrl+n/p, sometimes helix is blocked(any input)
> 
> Well I'm not sure where the error originates, but I suspect it doesn't have to do with this PR (can you try the same config on master?).
> 
> Regarding the blocking, can you try to comment out this block? If this helps, I have to think about a different way to sequentialize the LSP requests

I checked, it's a master problem.